api/searchCases/index.ts

```diff
@@ -1,137 +1,166 @@
+import { UserInfo } from '../auth/interfaces/UserInfo'
+import { getConfigValue } from '../configuration'
 import { caseMetaDataFiledsMapping } from '../configuration/mappings'
+import { WILDCARD_SEARCH_FIELDS, WILDCARD_SEARCH_ROLES } from '../configuration/references'
 import { fieldNameMapper } from '../lib/util'
+import { ElasticSearchQuery } from './interfaces/ElasticSearchQuery'
 
 /**
  * Manually creating Elastic search query
  */
 export function modifyRequest(proxyReq, req) {
-    const request = prepareElasticQuery(req.query, req.body)
+  const userInfo = req.session.passport.user.userinfo as UserInfo
+  const request = prepareElasticQuery(req.query, req.body, userInfo)
 
-    // Write out body changes to the proxyReq stream
-    const body = JSON.stringify(request)
+  // Write out body changes to the proxyReq stream
+  const body = JSON.stringify(request)
 
-    // Update header
-    proxyReq.setHeader('content-type', 'application/json')
-    proxyReq.setHeader('content-length', body.length)
+  // Update header
+  proxyReq.setHeader('content-type', 'application/json')
+  proxyReq.setHeader('content-length', body.length)
 
-    // Write out body changes to the proxyReq stream
-    proxyReq.write(body)
+  // Write out body changes to the proxyReq stream
+  proxyReq.write(body)
 
-    // Remove body-parser body object from the request
-    delete req.body
-    proxyReq.end()
+  // Remove body-parser body object from the request
+  delete req.body
+  proxyReq.end()
 }
 
-export function prepareElasticQuery(queryParams: {page?}, body: any): {} {
-    const metaCriteria = queryParams
-    let caseCriteria = {}
-    let nativeEsQuery: {} = {}
-    const matchList: any[] = []
-    const size = body.size || 10
-    const sort = body.sort ? prepareSort(body.sort) : []
-    const page = (queryParams.page || 1) - 1
-    const from = page * size
-
-    Object.keys(metaCriteria).map(key => {
-        if (key === 'ctid' || key === 'use_case' || key === 'view' || key === 'page') {
-            delete metaCriteria[key]
-        }
-
-        if (key.indexOf('case.') > -1) {
-            const newKey = key.replace('case.', '')
-            caseCriteria = {
-                ...caseCriteria,
-                [newKey]: metaCriteria[key],
-            }
-            delete metaCriteria[key]
-        }
-    })
+export function userCanPerformWildCardSearch(userInfo: UserInfo): boolean {
+  const allowedRoles = getConfigValue(WILDCARD_SEARCH_ROLES) as string[]
+  return userInfo.roles.filter((role: string) => allowedRoles
+    .map((allowedRole: string) => allowedRole.toLowerCase())
+    .indexOf(role.toLowerCase()) >= 0).length > 0
```

Comment:
Is `indexOf(...) >= 0` the desired approach? That means that `'caseworker'` can appear _anywhere_ in the string. Is that what they're after or should it be a prefix? Just a question, not saying it's wrong...

Reply:
Yes thats exactly what I want. The user should just have the role required to perform a wildcard search